o.js

```diff
@@ -168,19 +168,31 @@ o.prototype.ref = function (oRefInstance){
     return this;
 }
 
-o.prototype.push = function() {
-    return this.element;
-};
-
-o.prototype.logger = function() {
-    // console.log(this.element)
-};
-
 o.prototype.style = function(styles) {
     this.element.setAttribute('style', styles);
     return this;
 };
 
+// INPUT functions
+
+o.prototype.placeholder = function(placeholder) {
+    if(this.element.nodeName !== 'INPUT' || placeholder === undefined) {
+        console.warn('oJS: Cannot set placeholder. Element nodeName is not INPUT or argument given is undefined');
```

Comment:
może throw new Error zamiast console.warn? (do przegadania)

Reply:
Nie chciałem robić throw, żeby kod i tak działał. Bo jak damy .placeholder().anotherFn() to się wyłoży kod, a tak tylko placeholder się nie doda